src/data_types/strs.rs

```diff
@@ -567,21 +630,41 @@ mod tests {
 
     #[test]
     fn test_compare() {
-        let input: &CStr16 = cstr16!("test");
-
-        // test various comparisons with different order (left, right)
-        assert!(input.eq_str_until_nul(&"test"));
-        assert!(input.eq_str_until_nul(&String::from("test")));
-
-        // now other direction
-        assert!(String::from("test").eq_str_until_nul(input));
-        assert!("test".eq_str_until_nul(input));
-
-        // some more tests
-        // this is fine: compare until the first null
-        assert!(input.eq_str_until_nul(&"te\0st"));
-        // this is fine
-        assert!(input.eq_str_until_nul(&"test\0"));
-        assert!(!input.eq_str_until_nul(&"hello"));
+        // twice the same test: once for CStr8, once for CStr16
+        {
+            // test various comparisons with different order (left, right)
+            let input: &CStr8 = cstr8!("test");
+            assert!(input.eq_str_until_nul(&"test"));
+            assert!(input.eq_str_until_nul(&String::from("test")));
+
+            // now other direction
+            assert!(String::from("test").eq_str_until_nul(input));
+            assert!("test".eq_str_until_nul(input));
+
+            // some more tests
+            // this is fine: compare until the first null
+            assert!(input.eq_str_until_nul(&"te\0st"));
+            // this is fine
+            assert!(input.eq_str_until_nul(&"test\0"));
+            assert!(!input.eq_str_until_nul(&"hello"));
+        }
+        {
+            let input: &CStr16 = cstr16!("test");
+
+            // test various comparisons with different order (left, right)
+            assert!(input.eq_str_until_nul(&"test"));
+            assert!(input.eq_str_until_nul(&String::from("test")));
+
+            // now other direction
+            assert!(String::from("test").eq_str_until_nul(input));
+            assert!("test".eq_str_until_nul(input));
+
+            // some more tests
+            // this is fine: compare until the first null
+            assert!(input.eq_str_until_nul(&"te\0st"));
+            // this is fine
+            assert!(input.eq_str_until_nul(&"test\0"));
+            assert!(!input.eq_str_until_nul(&"hello"));
+        }
```

Comment:
Nit: could split this into two `#[test]` functions rather than the interior blocks.